tests/unit_tests/service/test_rest_api.py

```diff
@@ -171,19 +172,25 @@ class MyModel(BaseModel):
 
     plan = Plan(name="my-plan", model=MyModel)
     get_plan_mock.return_value = PlanModel.from_plan(plan)
-    submit_task_mock.side_effect = ValidationError.from_exception_data(
-        title="ValueError",
-        line_errors=[
-            InitErrorDetails(
-                type="missing", loc=("id",), msg="value is required for Identifier"
-            )  # type: ignore
-        ],
-    )
+
+    def raise_validation_error(bar: Any):
+        return ValidationError.from_exception_data(
+            title="ValueError",
+            line_errors=[
+                InitErrorDetails(
+                    input=bar,
+                    type="missing",
+                    loc=("id",),
+                )
+            ],
+        )
+
+    submit_task_mock.side_effect = raise_validation_error
     response = client.post("/tasks", json={"name": "my-plan"})
     assert response.status_code == 422
     assert response.json() == {
         "detail": (
-            "\n        Input validation failed: id: Field required,\n"
+            "\n        Input validation failed: task_id: Input should be a valid string,\n"
```

Comment:
I am not a fan of this change to the test- task_id name hasn't changed in this patchset.

However, InitErrorDetails.msg is not a field, and InitErrorDetails is a TypedDict- it should not allow extra fields as is being done here. The msg that was passed doesn't appear to have any impact on the message that is being returned.

Reply:
Resolved: the side effect wasn't actually raising the Exception that was created, either before or after the change.